ethereum/blockcreation/src/main/java/org/hyperledger/besu/ethereum/blockcreation/txselection/BlockTransactionSelector.java

```diff
@@ -437,17 +421,51 @@ private TransactionSelectionResult handleTransactionNotSelected(
 
     final var pendingTransaction = evaluationContext.getPendingTransaction();
 
-    transactionSelectionResults.updateNotSelected(
-        evaluationContext.getTransaction(), selectionResult);
-    pluginTransactionSelector.onTransactionNotSelected(evaluationContext, selectionResult);
+    final TransactionSelectionResult actualResult =
+        isTimeout.get() && !wasTimeoutAlreadyProcessed(selectionResult)
+            ? transactionTookTooLong(evaluationContext)
+                ? TX_EVALUATION_TOO_LONG
+                : BLOCK_SELECTION_TIMEOUT
+            : selectionResult;
+
+    transactionSelectionResults.updateNotSelected(evaluationContext.getTransaction(), actualResult);
+    pluginTransactionSelector.onTransactionNotSelected(evaluationContext, actualResult);
     LOG.atTrace()
-        .setMessage("Not selected {} for block creation with result {}, evaluated in {}")
+        .setMessage(
+            "Not selected {} for block creation with result {} (original result {}), evaluated in {}")
         .addArgument(pendingTransaction::toTraceLog)
+        .addArgument(actualResult)
         .addArgument(selectionResult)
         .addArgument(evaluationContext.getEvaluationTimer())
         .log();
 
-    return selectionResult;
+    return actualResult;
+  }
+
+  private boolean wasTimeoutAlreadyProcessed(final TransactionSelectionResult selectionResult) {
+    return selectionResult.equals(TX_EVALUATION_TOO_LONG)
+        || selectionResult.equals(BLOCK_SELECTION_TIMEOUT);
+  }
+
+  private boolean transactionTookTooLong(final TransactionEvaluationContext evaluationContext) {
+    final var evaluationTimer = evaluationContext.getEvaluationTimer();
+    if (evaluationTimer.elapsed(TimeUnit.MILLISECONDS) > blockTxsSelectionMaxTime) {
+      LOG.atWarn()
+          .setMessage(
+              "Transaction {} is too late for inclusion, evaluated in {} that is over the max limit of {}ms"
+                  + ", removing it from the pool")
+          .addArgument(evaluationContext.getPendingTransaction()::toTraceLog)
```

Comment:
is toTraceLog() at warn level too verbose?

Reply:
what does this log line look like in reality?

Reply:
Like that, it should not be frequent, do you think it is better to only print the hash?
```
Transaction 0xee8fa4f7c1aa6ce9e06b70d7fb1984934fc9f09614f53f3ff26fd87874797d92={MessageCall, 386, 0x7be0af0785193c5b00672c70364d545aaf35b3d6, EIP1559, mf: 4.00 gwei, pf: 3.00 gwei, gl: 7532434, v: 0 wei, to: 0x69d35b2d44442824e6aa105674779eb8e5091cd2} is too late for inclusion, evaluated in 2.256 s that is over the max limit of 1800ms, removing it from the pool
```

Reply:
if it's not too frequent it's prob ok 👍